### PR TITLE
[build] add `-data` flag to slc calls

### PR DESCRIPTION
**Pull request by lmnotran**
This fixes concurrency issues in CI where multiple SLC instances may be running, where each may be using a different sdk

### CODE REVIEW

**Review by romacdon — APPROVED** [review]
LGTM

File: script/build_example_apps [hunk 1]
```diff
@@ -59,6 +59,7 @@ readonly OT_OPTIONS=(
     "-DOT_EXTERNAL_HEAP=ON"
     "-DOT_PING_SENDER=ON"
     "-DOT_SLAAC=ON"
+    "-DOT_PING_SENDER=ON"
```
[the remark this answers is not in the document]
Comment (romacdon) [reply script/build_example_apps]: This is resolved.

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Review by parag-silabs — APPROVED** [review]

**Comment by jwhui** [comment]
@lmnotran , build checks are failing. Is this expected?

**Comment by lmnotran** [comment]
@jwhui Can you take a look at the PR when you get a chance? I just fixed the build checks

**Review by jwhui — APPROVED** [review]

**Pull request merged**